cirq-core/cirq/qis/clifford_tableau_test.py

```diff
@@ -322,3 +322,91 @@ def test_deprecated_clifford_location():
         from cirq.sim import CliffordTableau
 
         CliffordTableau(num_qubits=1)
+
+
+def test_merge_tableau():
+    def three_identical_table(num_qubits):
+        t1 = cirq.CliffordTableau(num_qubits)
+        t2 = cirq.CliffordTableau(num_qubits)
+        t3 = cirq.CliffordTableau(num_qubits)
+        return t1, t2, t3
+
+    t1, t2, expected_t = three_identical_table(1)
+    assert expected_t == t1.merged_with(t2)
+
+    t1, t2, expected_t = three_identical_table(1)
+    _ = [_H(t, 0) for t in (t1, expected_t)]
+    _ = [_H(t, 0) for t in (t2, expected_t)]
+    assert expected_t == t1.merged_with(t2)
+
+    t1, t2, expected_t = three_identical_table(1)
+    _ = [_X(t, 0) for t in (t1, expected_t)]
+    _ = [_S(t, 0) for t in (t2, expected_t)]
+    assert expected_t == t1.merged_with(t2)
+
+    t1, t2, expected_t = three_identical_table(1)
+    _ = [_X(t, 0) for t in (t1, expected_t)]
+    _ = [_H(t, 0) for t in (t1, expected_t)]
+    _ = [_Z(t, 0) for t in (t1, expected_t)]
+    _ = [_S(t, 0) for t in (t2, expected_t)]
+    _ = [_H(t, 0) for t in (t2, expected_t)]
+    assert expected_t == t1.merged_with(t2)
```

Comment:
It's worth adding a test case that tests `expected_t != t1.merged_with(t2)` when `t2` is acting before `t1` i.e. Enforce the order of composition.

Reply:
Yes, that sounds good